src/services/dataService.js

```diff
@@ -0,0 +1,67 @@
+import * as axios from 'axios';
+import {
+  API_AUTH_URL,
+  API_BASE_URL,
+  DATA_TYPES,
+  API_SEARCH_LOCATION_DATA_TYPE,
+  API_GET_CURRENT_WEATHER_DATA_TYPE,
+  API_GET_FORECAST_DATA_TYPE
+} from '../constants/constants';
+
+const instance = axios.create({
+  withCredentials: true,
+  baseURL: API_BASE_URL
+});
+
+export const weatherAPI = {
+  token: null,
+  getAuthHeaders() {
+    return {
+      headers: {
+        Authorization: `Bearer ${this.token}`
+      }
+    };
+  },
+  async getToken(user, password) {
+    const body = {
+      user: `${user}`,
+      password: `${password}`
+    };
+    try {
+      const response = await instance.post(API_AUTH_URL, body, {
+        headers: {
+          'Content-Type': 'text/plain'
+        }
+      });
+      this.token = response.data.access_token;
+    } catch (error) {
+      console.log(error);
+      this.token = null;
+    }
+
+    return this.token;
+  },
+
+  async getData(dataType, param) {
```

Comment:
what if there are more params?

Reply:
getData is general function, it fetches different variants of data form url with param. If exact getting data function (such as getForecast) needs some more params it concats all the params into one string param and call getData with this concated param.
If it's bad idea please advise how to do this in better way.

Reply:
To my mind this is an overcomplication and overengeneering: I'd say to just have separate functions like that: 
```
async searchLocation(query) {
    ...
    try {
      const response = await instance.get(...here you will pass url, query and other stuff within axios config);
      return response.data[dataKey];
    } catch (error) {
      console.error(error);
    }
},

etc
```

Reply:
thanks, changed